crates/notary/server/src/settings.rs

```
@@ -0,0 +1,50 @@
+use config::{Config, ConfigError, Environment, File};
+use std::path::Path;
+use tracing::{info, warn, debug};
+use crate::{CliFields, NotaryServerProperties};
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub struct Settings {
+    #[serde(flatten)]
+    pub config: NotaryServerProperties,
+}
+
+impl Settings {
+    pub fn new(cli_fields: &CliFields) -> Result<Self, ConfigError> {
+        let mut builder = Config::builder();
+
+        // Add the config file if it exists
+        let config_path = Path::new(&cli_fields.config_file);
+
+        if config_path.exists() {
+            info!("Loading configuration from: {}", cli_fields.config_file);
+            builder = builder.add_source(File::from(config_path));
+        } else {
+            warn!("Config file not found: {}. Using defaults and overrides.", cli_fields.config_file);

```

should throw an error if config file is not found